google/datalab/_context.py

```diff
@@ -112,9 +112,15 @@ def default():
     Returns:
       An initialized and shared instance of a Context object.
     """
-    if Context._global_context is None:
-      project = get_default_project_id()
-      credentials = get_credentials()
-      config = Context._get_default_config()
+    credentials = get_credentials()
+    project = get_default_project_id()
+    config = Context._get_default_config()
```

Comment:
This will wipe out any modifications a user has made to the default config with a static dictionary.

I don't think that is what we want.

I would just move the config setting back inside of the then branch for the subsequent if statement.